frigate/object_detection.py

```diff
@@ -22,46 +29,150 @@ def ReadLabelFile(file_path):
-        self.labels = ReadLabelFile(PATH_TO_LABELS)
+        if tf_args != None:
+            tf_model = tf_args.get('model', None)
+            tf_labels = tf_args.get('labels', None)
+            self.data_dir = tf_args.get('data_dir', DATA_DIR)
+        else:
+            tf_model = PATH_TO_CKPT
+            tf_labels = PATH_TO_LABELS
+            self.data_dir = DATA_DIR
+        print("Loading Tensorflow model: " + str(tf_model))
+        self.engine = DetectionEngine(tf_model)
+        if tf_labels:
+            self.labels = ReadLabelFile(tf_labels)
+        else:
+            self.labels = None
+        yesterday = datetime.datetime.now() - datetime.timedelta(days=1)
+        self.last_saved_det_frame_time = yesterday
+        self.last_saved_non_det_frame_time = yesterday
 
     def run(self):
         # process queue...
         while True:
             frame = self.prepped_frame_queue.get()
 
             # Actual detection.
-            objects = self.engine.DetectWithInputTensor(frame['frame'], threshold=frame['region_threshold'], top_k=3)
+#            objects = self.engine.DetectWithInputTensor(frame['frame'], threshold=frame['region_threshold'], top_k=3)
             # print(self.engine.get_inference_time())
 
-            # parse and pass detected objects back to the camera
-            parsed_objects = []
-            for obj in objects:
-                box = obj.bounding_box.flatten().tolist()
-                parsed_objects.append({
-                            'frame_time': frame['frame_time'],
-                            'name': str(self.labels[obj.label_id]),
-                            'score': float(obj.score),
-                            'xmin': int((box[0] * frame['region_size']) + frame['region_x_offset']),
-                            'ymin': int((box[1] * frame['region_size']) + frame['region_y_offset']),
-                            'xmax': int((box[2] * frame['region_size']) + frame['region_x_offset']),
-                            'ymax': int((box[3] * frame['region_size']) + frame['region_y_offset'])
-                        })
-            self.cameras[frame['camera_name']].add_objects(parsed_objects)
+            try:
+                # print("Calling TF Coral to detect objects")
+                detected_objects = self.engine.DetectWithImage(
+                    frame['img'],
+                    threshold=frame['region_threshold'],
+                    keep_aspect_ratio=True,
+                    relative_coord=True,
+                    top_k=3)
+                #print("TF Coral detected {} objects withing {} ms".
+                #      format(len(objects),
+                #             self.engine.get_inference_time()))
+
+                # parse and pass detected objects back to the outbound camera stream
+                self.__pass_detections_to_output_cam(frame, detected_objects)
+
+                # save detections to file if enabled
+                self.__save_sample(frame, detected_objects)
```

Comment:
Because this is done on the same thread as the Coral, it must finish saving the files to disk before it can begin working on the next image. That will also reduce the FPS of the Coral.